clang/lib/ExtractAPI/Serialization/SymbolGraphSerializer.cpp

```diff
@@ -815,33 +815,25 @@ void SymbolGraphSerializer::serializeTypedefRecord(
 
 Object SymbolGraphSerializer::serialize() {
   // Serialize global variables in the API set.
-  for (const auto &GlobalVar : API.getGlobalVariables())
-    serializeGlobalVariableRecord(*GlobalVar.second);
+  this->traverseGlobalVariableRecords();
```

Comment:
This defines the traversal logic for a whole APISet and should be moved into the base class. Additionaly the `this->` is superfluous and we don't tend to use that syntax when not needed in LLVM. (However moving this into the base class would mean you would need to do the whole `static_cast<Derived *>(this)` thing`